### PR TITLE
common: assert(0) -> ceph_abort()

### PR DESCRIPTION
**Pull request by liewegas**
This converts all assert(0) instances. We still need to fix assert(0 == "something") callers.

### CODE REVIEW

File: src/mon/MonitorDBStore.h [hunk 1]
```diff
@@ -290,7 +290,7 @@ class MonitorDBStore
 	break;
       default:
 	derr << __func__ << " unknown op type " << op.type << dendl;
-	ceph_assert(0);
+	ceph_ceph_abort();
```
Comment (xiexingguo) [inline src/mon/MonitorDBStore.h]: s/ceph_ceph_abort/ceph_abort/

**Comment by liewegas** [comment]
fixed

File: src/test/on_exit.cc [hunk 2]
```diff
@@ -106,7 +106,7 @@ int main(int argc, char **argv)
     *new_val = EXIT_FUNC_VAL;
     exit_func_mgr.add_callback(exit_func_cb, new_val);
     call_exit();
-    assert(0);
+    ceph_abort();
```
Comment (xiexingguo) [inline src/test/on_exit.cc]: [ 64%] Building CXX object src/test/CMakeFiles/unittest_on_exit.dir/on_exit.cc.o
/home/jenkins-build/build/workspace/ceph-pull-requests/src/test/on_exit.cc: In function ‘int main(int, char**)’:
/home/jenkins-build/build/workspace/ceph-pull-requests/src/test/on_exit.cc:109:16: error: ‘ceph_abort’ was not declared in this scope
     ceph_abort();

**Comment by xiexingguo** [comment]
still does not compile

File: src/os/bluestore/BitAllocator.cc [hunk 3]
```diff
@@ -379,18 +379,18 @@ int64_t BitMapZone::get_used_blocks()
 
 bool BitMapZone::reserve_blocks(int64_t num_blocks)
 {
-  alloc_assert(0);
+  alloc_ceph_abort();
```
Comment (xiexingguo) [inline src/os/bluestore/BitAllocator.cc]: We shall withdraw these changes and change 
https://github.com/ceph/ceph/blob/master/src/os/bluestore/BitAllocator.h#L21 instead.

**Comment by liewegas** [comment]
I think we still want ot change whatever_assert(0) -> ceph_abort().

**Comment by liewegas** [comment]
compiles now

**Comment by liewegas** [comment]
rebased

**Comment by dachary** [comment]
jenkins test this please (eio)

**Comment by dachary** [comment]
jenkins test this please (eio)

**Comment by tchaikov** [comment]
@liewega needs rebase.

**Comment by tchaikov** [comment]
```
In file included from /srv/autobuild-ceph/gitbuilder.git/build/out~/ceph-11.0.2-1849-gcdc9f9d/src/mgr/PyModules.cc:21:0:
/srv/autobuild-ceph/gitbuilder.git/build/out~/ceph-11.0.2-1849-gcdc9f9d/src/mgr/PyFormatter.h: In member function ‘virtual void PyFormatter::open_array_section_in_ns(const char*, const char*)’:
error: /srv/autobuild-ceph/gitbuilder.git/build/out~/ceph-11.0.2-1849-gcdc9f9d/src/mgr/PyFormatter.h:57:15: ‘ceph_abort’ was not declared in this scope
{ceph_abort();}
^
```
see http://gitbuilder.sepia.ceph.com/gitbuilder-ceph-deb-trusty-amd64-notcmalloc/log.cgi?log=cdc9f9d1280f09fbd3e2b6a047cc7df41a94b870

testing with https://github.com/ceph/ceph/commit/0b77c6bbe8e5d7e5772c02eb74f64e559544ef3c

**Comment by tchaikov** [comment]
http://pulpito.ceph.com/kchai-2016-11-22_09:44:32-rados-wip-kefu-testing---basic-mira/

**Pull request merged**